automation/taskcluster/androidTest/ui-test.sh

```diff
@@ -74,19 +77,48 @@ else
     flank_template="$FLANK_CONF_ARM"
 fi
 
-APK_APP="./samples/${component}/build/outputs/apk/geckoNightly/debug/samples-browser-geckoNightly-debug.apk"
-APK_TEST="./components/${component}/engine-gecko-nightly/build/outputs/apk/androidTest/debug/browser-engine-gecko-nightly-debug-androidTest.apk"
+# Remove samples- from the component for each APK path
+samples=${component//samples-}
 
+# If tests are for components, the path is different than for samples
```

Comment:
This is great!  My only suggestion is to word the comments a bit clearer.  Perhaps something like:
```
# Case 1: tests for any component (but NOT sample, NOT real UI tests)
# Case 2: tests for browser sample (geckoNightly sample only)
# Case 3:  tests for non-browser samples (i.e.  samples-glean)
```
Hopefully I understood your comment for case #2 correctly